### PR TITLE
[FEAT] optimize IAB init

### PR DESCRIPTION
**Pull request by JohnathanWhite**
- refactored IAB ping method to postMessage from its success callback
- synced theme change with app
- added loading wrapper to prevent flashing of initial theme load
- refactored and synced card advertisements with card initial fetch, connecting/disconnecting bitpay id events

### CODE REVIEW

**Comment by cnguyen812** [comment]
ACK

File: src/providers/gift-card/gift-card.ts [hunk 1]
```diff
@@ -67,7 +67,6 @@ export class GiftCardProvider extends InvoiceProvider {
   ) {
     super(emailNotificationsProvider, http, logger, persistenceProvider);
     this.logger.debug('GiftCardProvider initialized');
-    this.setCredentials();
```
Comment (cmgustavo) [inline src/providers/gift-card/gift-card.ts:70]: Not sure about this. `setCredentials` exists on extended class InvoiceProvider.

Comment (cmgustavo) [reply src/providers/gift-card/gift-card.ts:70]: Oh, it was removed below. 👍

**Review by cmgustavo — APPROVED** [review]
ACK! 🚀

**Pull request merged**